File: app/src/main/java/org/stepic/droid/store/CancelSniffer.java
package org.stepic.droid.store;

public interface CancelSniffer {
    void addStepIdCancel(long stepId);

    void removeStepIdCancel(long stepId);

    boolean isStepIdCanceled(long stepId);

    void addSectionIdCancel(long sectionId);

    void removeSectionIdCancel(long sectionId);

    boolean isSectionIdIsCanceled(long sectionId);

    void addLessonToCancel(long lessonId);

    void removeLessonIdToCancel(long lessonId);

    boolean isLessonIdIsCanceled(long lessonId);
}

File: app/src/main/java/org/stepic/droid/store/CancelSniffer.kt
package org.stepic.droid.store

interface CancelSniffer {
    fun addStepIdCancel(stepId: Long)

    fun removeStepIdCancel(stepId: Long)

    fun isStepIdCanceled(stepId: Long): Boolean

    fun addSectionIdCancel(sectionId: Long)

    fun removeSectionIdCancel(sectionId: Long)

    fun isSectionIdCanceled(sectionId: Long): Boolean
}
